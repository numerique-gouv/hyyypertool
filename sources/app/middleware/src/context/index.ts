//

export * from "./set_variables";
