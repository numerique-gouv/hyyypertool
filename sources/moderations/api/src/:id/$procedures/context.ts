//

import type { App_Context } from "@~/app.middleware/context";
import type { Crisp_Context } from "@~/crisp.middleware";

//

export type ContextType = App_Context & Crisp_Context;
