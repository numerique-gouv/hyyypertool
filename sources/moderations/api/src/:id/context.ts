//

import type { App_Context } from "@~/app.middleware/context";
import { urls } from "@~/app.urls";
import type { GetModerationWithDetailsDto } from "@~/moderations.repository";
import type { GetFicheOrganizationByIdHandler } from "@~/organizations.lib/usecase";
import {
  type GetDomainCountDto,
  type GetOrganizationMemberDto,
  type GetOrganizationMembersCountDto,
} from "@~/organizations.repository";
import type { Env, InferRequestType } from "hono";
import { useRequestContext } from "hono/jsx-renderer";

//

export interface ModerationContext extends Env {
  Variables: {
    moderation: GetModerationWithDetailsDto;
  };
}
export interface ContextVariablesType extends Env {
  Variables: {
    domain: string;
    moderation: GetModerationWithDetailsDto;
    organization_member: GetOrganizationMemberDto;
    organization_fiche: Awaited<ReturnType<GetFicheOrganizationByIdHandler>>;
    query_organization_members_count: Promise<GetOrganizationMembersCountDto>;
    query_domain_count: Promise<GetDomainCountDto>;
  };
}
export type ContextType = App_Context & ContextVariablesType;

//

const $get = typeof urls.moderations[":id"].$get;

type PageInputType = {
  out: InferRequestType<typeof $get>;
};

export const usePageRequestContext = useRequestContext<
  ContextType,
  any,
  PageInputType
>;

//
