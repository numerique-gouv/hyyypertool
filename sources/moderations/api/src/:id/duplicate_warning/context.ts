//

import { Entity_Schema } from "@~/app.core/schema";
import type { App_Context } from "@~/app.middleware/context";
import type { IdentiteProconnect_PgDatabase } from "@~/identite-proconnect.database";
import { Duplicate_Warning } from "./Duplicate_Warning";
import type { Env } from "hono";
import { useRequestContext } from "hono/jsx-renderer";
import { z } from "zod";

//

export async function loadDuplicateWarningPageVariables(
  pg: IdentiteProconnect_PgDatabase,
  { moderation_id, organization_id, user_id }: { 
    moderation_id: number; 
    organization_id: number; 
    user_id: number; 
  },
) {
  const value = await Duplicate_Warning.queryContextValues(pg, {
    moderation_id,
    organization_id,
    user_id,
  });

  return value;
}

//

export interface ContextVariablesType extends Env {
  Variables: Awaited<ReturnType<typeof loadDuplicateWarningPageVariables>>;
}
export type ContextType = App_Context & ContextVariablesType;

//

export const QuerySchema = z.object({
  organization_id: z.string().pipe(z.coerce.number().int().nonnegative()),
  user_id: z.string().pipe(z.coerce.number().int().nonnegative()),
});

export const ParamSchema = Entity_Schema;

//

type PageInputType = {
  out: {
    param: z.input<typeof ParamSchema>;
    query: z.input<typeof QuerySchema>;
  };
};

export const usePageRequestContext = useRequestContext<
  ContextType,
  any,
  PageInputType
>;