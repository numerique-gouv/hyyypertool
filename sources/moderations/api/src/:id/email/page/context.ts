//

import { DescribedBy_Schema, Entity_Schema } from "@~/app.core/schema";
import type { App_Context } from "@~/app.middleware/context";
import { get_crisp_mail } from "@~/crisp.lib";
import type { Crisp_Context } from "@~/crisp.middleware";
import type { IdentiteProconnect_PgDatabase } from "@~/identite-proconnect.database";
import { GetCripsFromSessionId } from "@~/moderations.lib/usecase/GetCripsFromSessionId";
import { GetZammadFromTicketId } from "@~/moderations.lib/usecase/GetZammadFromTicketId";
import { GetModerationForEmail } from "@~/moderations.repository";
import { get_zammad_mail } from "@~/zammad.lib/get_zammad_mail";
import { to } from "await-to-js";
import { type Env } from "hono";
import { useRequestContext } from "hono/jsx-renderer";
import type { z } from "zod";

//

export async function loadEmailPageVariables(
  pg: IdentiteProconnect_PgDatabase,
  { id, crisp_config }: { id: number; crisp_config: any },
) {
  const MAX_ARTICLE_COUNT = 3;

  // Load moderation data
  const get_moderation_for_email = GetModerationForEmail(pg);
  const moderation = await get_moderation_for_email(id);

  // Load zammad data if ticket_id exists
  if (moderation.ticket_id) {
    const get_zammad_from_ticket_id = GetZammadFromTicketId({
      fetch_zammad_mail: get_zammad_mail,
    });
    const [, zammad_result] = await to(
      get_zammad_from_ticket_id({
        ticket_id: moderation.ticket_id,
        limit: MAX_ARTICLE_COUNT,
      }),
    );
    return {
      MAX_ARTICLE_COUNT,
      moderation,
      zammad: zammad_result,
    };
  }

  // Load crisp data if session_id exists (using ticket_id as session_id)
  if (moderation.ticket_id) {
    const get_crisp_from_session_id = GetCripsFromSessionId({
      crisp_config,
      fetch_crisp_mail: get_crisp_mail,
    });
    const [, crisp_result] = await to(
      get_crisp_from_session_id({
        session_id: moderation.ticket_id,
        limit: MAX_ARTICLE_COUNT,
      }),
    );
    return {
      MAX_ARTICLE_COUNT,
      moderation,
      crisp: crisp_result,
    };
  }

  return {
    MAX_ARTICLE_COUNT,
    moderation,
  };
}

//

export interface ContextVariablesType extends Env {
  Variables: Awaited<ReturnType<typeof loadEmailPageVariables>>;
}
export type ContextType = App_Context & Crisp_Context & ContextVariablesType;

//

type PageInputType = {
  out: {
    param: z.input<typeof Entity_Schema>;
    query: z.input<typeof DescribedBy_Schema>;
  };
};

export const usePageRequestContext = useRequestContext<
  ContextType,
  any,
  PageInputType
>;
