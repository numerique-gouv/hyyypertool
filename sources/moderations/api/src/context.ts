//

import { Pagination_Schema, type Pagination } from "@~/app.core/schema";
import { z_coerce_boolean } from "@~/app.core/schema/z_coerce_boolean";
import { z_empty_string_to_undefined } from "@~/app.core/schema/z_empty_string_to_undefined";
import type { get_moderations_list } from "@~/moderations.repository/get_moderations_list";
import { createContext } from "hono/jsx";
import { z } from "zod";

//

export const MODERATION_TABLE_ID = "moderation_table";
export const MODERATION_TABLE_PAGE_ID = "moderation_table_page";

export const Search_Schema = z.object({
  day: z
    .string()
    .default("")
    .pipe(z.coerce.date().or(z_empty_string_to_undefined)),
  search_siret: z.string().default(""),
  search_email: z.string().default(""),
  processed_requests: z.string().pipe(z_coerce_boolean).default("false"),
  hide_non_verified_domain: z.string().pipe(z_coerce_boolean).default("false"),
  hide_join_organization: z.string().pipe(z_coerce_boolean).default("false"),
});
export const Page_Query = Search_Schema.merge(Pagination_Schema).partial();
export type Search = z.infer<typeof Search_Schema>;

//

export default createContext({
  query_moderations_list: {} as Promise<get_moderations_list_dto>,
  pagination: {} as Pagination,
});

export type get_moderations_list_dto = Awaited<
  ReturnType<typeof get_moderations_list>
>;
