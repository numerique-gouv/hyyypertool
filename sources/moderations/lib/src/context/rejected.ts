//

import type { AgentConnect_UserInfo } from "@~/app.middleware/session";
import type { Config } from "@~/crisp.lib/types";
import type { IdentiteProconnect_PgDatabase } from "@~/identite-proconnect.database";
import type { GetModerationDto } from "@~/moderations.repository";
import type { RejectedMessage } from "../schema/rejected.form";

//

export type RejectedModeration_Context = {
  crisp_config: Config;
  moderation: GetModerationDto;
  pg: IdentiteProconnect_PgDatabase;
  reason: string;
  resolve_delay: number;
  subject: string;
  userinfo: AgentConnect_UserInfo;
};
export type RejectedFullMessage = RejectedMessage & { to: string };
