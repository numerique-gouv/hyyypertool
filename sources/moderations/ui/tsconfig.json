{
  "compilerOptions": {
    "jsxImportSource": "hono/jsx",
    "outDir": "./node_modules/.cache/tsc"
  },
  "extends": "@~/config.typescript/api/tsconfig.json",
  "include": ["src"],
  "references": [
    {
      "path": "../../app/ui"
    },
    {
      "path": "../../app/urls"
    },
    {
      "path": "../lib"
    },
    {
      "path": "../../organizations/lib"
    },
    {
      "path": "../../users/lib"
    },
    {
      "path": "../../infra/crisp/lib"
    }
  ]
}
