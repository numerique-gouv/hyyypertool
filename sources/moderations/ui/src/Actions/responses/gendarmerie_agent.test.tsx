//

import { render_md } from "@~/app.ui/testing";
import { expect, test } from "bun:test";
import { context, type Values } from "../context";
import gendarmerie_agent from "./gendarmerie_agent";

//

test("returns error on 🦄 organization (gendarmerie) email", async () => {
  expect(
    await render_md(
      <context.Provider
        value={
          {
            moderation: {
              organization: {
                cached_libelle: "🦄",
                siret: "🌙✨",
                cached_libelle_categorie_juridique: "🐯",
              },
            },
          } as Values
        }
      >
        <Response />
      </context.Provider>,
    ),
  ).toMatchSnapshot();
});

function Response() {
  return <>{gendarmerie_agent()}</>;
}
