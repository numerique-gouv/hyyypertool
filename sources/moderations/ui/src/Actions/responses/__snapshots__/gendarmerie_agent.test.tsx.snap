// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`returns error on 🦄 organization (gendarmerie) email 1`] = `
"Bonjour,

Nous avons bien reçu votre demande de rattachement à l&#39;organisation « 🦄 » sur ProConnect (anciennement : AgentConnect, MonComptePro).

Pour information, vous semblez avoir fait erreur sur l&#39;organisation à laquelle vous souhaitez rattacher votre compte utilisateur ProConnect.
L&#39;organisation « 🦄 » enregistrée sous le SIRET 🌙✨ est une « 🐯 ».
Pour information, les numéros SIREN/SIRET des organisations publiques commencent obligatoirement par 1 ou 2 (source : définition de l&#39;INSEE du Numéro SIREN).
Comme vous pouvez le constater, le SIRET de l’organisation que vous avez choisi ne correspond pas à une organisation publique.
Nous vous invitons donc à créer à nouveau votre compte utilisateur en le rattachant à l&#39;organisation publique dont vous dépendez (ex : région de gendarmerie dont vous dépendez…).

Nous restons à votre disposition pour tout complément d&#39;information.

Bien cordialement,
L’équipe ProConnect.
"
`;
